msteams-platform/tabs/how-to/authentication/tab-sso-graph-api.md

```diff
@@ -213,25 +213,30 @@ If you need to access Microsoft Graph data, configure your server-side code to:
+Another approach for getting Graph scopes is to present a consent dialog using our existing [third party OAuth provider authentication approach](~/tabs/how-to/authentication/auth-tab-aad.md#navigate-to-the-authorization-page-from-your-pop-up-page). This approach involves popping up an Azure AD consent dialog box.
+
+You can also ask for consent using the TeamsJS [authentication](/javascript/api/@microsoft/teams-js/authentication) capability. 
 
 <details>
-<summary>To ask for additional consent using the Auth API, follow these steps:</summary>
+<summary>To obtain consent using the TeamsJS client library, follow these steps:</summary>
+
+> [!TIP]
+> The [Teams Personal Tab SSO Authentication](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/tab-personal-sso-quickstart/js/src/components/Tab.js#L64-L101) sample provides code demonstrating following steps.
 
 1. The token retrieved using `getAuthToken()` must be exchanged on the server-side using Azure AD [on-behalf-of flow](/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow) to get access to those other Graph APIs. Ensure you use the v2 Graph endpoint for this exchange.
-2. If the exchange fails, Azure AD returns an invalid grant exception. It usually responds with one of the two error messages, `invalid_grant` or `interaction_required`.
-3. When the exchange fails, you must ask for consent. Use the user interface (UI) to ask the app user to grant other consent. This UI must include a button that triggers an Azure AD consent dialog using [Silent authentication](~/concepts/authentication/auth-silent-aad.md).
-4. When asking for more consent from Azure AD, you must include `prompt=consent` in your [query-string-parameter](~/tabs/how-to/authentication/auth-silent-aad.md#get-the-user-context) to Azure AD, otherwise Azure AD wouldn't ask for other scopes.
+2. Often the first time you try to execute this token exchange for a user, Azure AD will refuse to exchange tokens because the user has not consented to giving your app permission to their data for this purpose. In these cases, your exchange will fail with either the `invalid_grant` or `interaction_required` error.
+3. If the exchange fails for one of these two reasons, you must prompt the user for consent. Since user interaction can only happen from the client, your server will need to return an indication to your client app that consent is required. You can then use the user interface (UI) to ask the app user to grant other consent. This UI must include a button that triggers an Azure AD consent dialog using [Silent authentication](~/concepts/authentication/auth-silent-aad.md).
```

Comment:
Is this correct? I am, in no sense of the word, an expert here -- but conceptually the idea of "silent" authentication triggering a dialog feels like an oxymoron?

Reply:
Haha, fair point. I attempted to rework this and point to a current article (and screenshot): https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/tab-sso-code#consent-dialog-for-getting-access-token

Reply:
Hmmm.. I looked at the content on that page and I don't think it's helpful in this case. In this situation, I believe the developer needs to call [authentication.authenticate](https://learn.microsoft.com/en-us/javascript/api/@microsoft/teams-js/authentication?view=msteams-client-js-latest#@microsoft-teams-js-authentication-authenticate) and provide appropriate parameters to trigger the consent dialog -- notably crafting an AAD URL with the `prompt=consent` parameter on it. Not sure if there is an existing place that explains how to do this properly.

Reply:
I added this reference from AAD docs: https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow#send-the-sign-in-request